### PR TITLE
Various bug fixes for the Scene class

### PR DESCRIPTION
**Pull request by agorski3**

### CODE REVIEW

**Comment by mbenz89** [comment]
ok to test

File: src/main/java/soot/Scene.java [hunk 1]
```diff
@@ -947,12 +959,13 @@ public Type getType(String arg) {
         result = VoidType.v();
       } else if (type.equals("boolean")) {
         result = BooleanType.v();
-      } else {
-        throw new RuntimeException("unknown type: '" + type + "'");
+      } else if (allowsPhantomRefs() && phantomNonExist) {
```
Comment (mbenz89) [inline src/main/java/soot/Scene.java]: Could you please document the meaning of phantomNonExist in the JavaDoc of this method?

**Comment by agorski3** [comment]
All requested changes have been made @mbenz89

**Comment by mbenz89** [comment]
retest this please

**Pull request merged**